thorlcr/master/thgraphmanager.cpp

```diff
@@ -921,13 +922,17 @@ bool CJobManager::executeGraph(IConstWorkUnit &workunit, const char *graphName,
         updateWorkunitStat(wu, SSTgraph, graphName, StTimeElapsed, graphTimeStr, graphTimeNs, wfid);
 
         addTimeStamp(wu, SSTgraph, graphName, StWhenFinished, wfid);
-        
+
+        fatalHdlr.setown(job->clearFatalHandler());
+
         removeJob(*job);
     }
     catch (IException *e)
     {
         exception.setown(e);
     }
+    if (!fatalHdlr)
+        fatalHdlr.setown(job->clearFatalHandler());
```

Comment:
I think the best place to call this is just before the job.clear() (on line 943), i.e. just before it would destroy the fatalHandler, then if it hangs at any point , including when if gets struck during destruction, it will still kick in and terminate the process.

Reply:
ok, will move it there.

Reply:
But wasn't the issue that endJob() never returned ?

Reply:
ok, I remember now we removed the clear() in go(), thanks.